src/Equinox.Core/Infrastructure.fs

```diff
@@ -42,26 +43,27 @@ type Async with
     /// </summary>
     /// <param name="task">Task to be awaited.</param>
     [<DebuggerStepThrough>]
-    static member AwaitTaskCorrect(task : System.Threading.Tasks.Task<'T>) : Async<'T> =
-        Async.FromContinuations(fun (sc,ec,_) ->
-            task.ContinueWith(fun (t : System.Threading.Tasks.Task<'T>) ->
+    static member AwaitTaskCorrect(task : Task<'T>) : Async<'T> =
```

Comment:
these prefixings are intentional - the idea is to be able to transplant without context anywhere

Reply:
I looked at https://github.com/jet/propulsion/blob/master/src/Propulsion/Infrastructure.fs for this, which one would be the winner? :)

Reply:
oh, sorry :( The Propulsion one is wrong ;) (I think I let Rider win me over incorrectly)

I'm less convinced its critical (esp as I accepted the System.Diagnostics one)

Reply:
Sooo, should I let you sleep a night on it to come to terms with removing it and keeping the `open`? ;)

Reply:
Ha, I'm not going to steer you either way ;)